api/v1alpha2/linodeobjectstoragekey_types.go

```diff
@@ -52,8 +52,17 @@ type LinodeObjectStorageKeySpec struct {
 	// SecretType instructs the controller what type of secret to generate containing access key details.
 	// +kubebuilder:validation:Enum=Opaque;addons.cluster.x-k8s.io/resource-set
 	// +kubebuilder:default=Opaque
+	// +kubebuilder:validation:XValidation:rule="self == oldSelf",message="Value is immutable"
 	// +optional
 	SecretType corev1.SecretType `json:"secretType,omitempty"`
+
+	// SecretDataFormat instructs the controller how to format the data stored in the secret containing access key details.
+	// It supports Go template syntax and interpolating the following values: .AccessKey, .SecretKey.
+	// If no format is supplied then a generic one is used containing the values specified.
+	// When SecretType is set to addons.cluster.x-k8s.io/resource-set, a .BucketEndpoint value is also available pointing to the location of the first bucket specified in BucketAccess.
+	// +kubebuilder:validation:XValidation:rule="self == oldSelf",message="Value is immutable"
+	// +optional
```

Comment:
Do we want to remove this optional marker if it looks like we error if this being unset in https://github.com/linode/cluster-api-provider-linode/pull/476/files#diff-cd057278b04faf08bb16208c91ddf55e5777d6c317e376fae1e84e071d9e3199R123?

Reply:
We still want this to optional whenever the secret type is opaque. We only require templates whenever we are populating a cluster resource-set secret type.

Reply:
should we add a webhook to error out whenever this is required but not set?